contrib/question_converters/canvas/quiz2pl.py

```diff
@@ -102,6 +103,67 @@ def clean_question_text(text: str) -> str:
     return text
 
 
+def image_file_extension(content_type: str) -> str:
+    match content_type:
+        case "image/x-icon":
+            return "ico"
+        case "image/svg+xml":
+            return "svg"
+        case _:
+            return content_type.split("/")[1]
+
+
+def handle_images(question_dir: str, text: str) -> str:
+    # Links to images will still point to Canvas. We need to download them and
+    # replace them with `pl-figure` elements.
+    #
+    # We use regex instead of a proper HTML parser because we want to limit this
+    # script to only using the Python standard library.
+    image_idx = 1
+    for match in re.finditer(r'(<p>)?(<img[^>]*src="([^"]+)"[^>]*>)(</p>)?', text):
+        url = match.group(3)
+        if not url.startswith("http"):
+            continue
+
+        # Set up the `clientFilesQuestion` directory for this question.
+        client_files_question_dir = os.path.join(question_dir, "clientFilesQuestion")
+        os.makedirs(client_files_question_dir, exist_ok=True)
+
+        # Canvas image URLs don't include the file extension, so we need to
+        # extract it from the `Content-Type` header.
+        res = requests.get(url)
+        res.raise_for_status()
+        extension = image_file_extension(res.headers["Content-Type"])
+
+        file_name = f"image_{image_idx}.{extension}"
+        file_path = os.path.join(client_files_question_dir, file_name)
+        with open(file_path, "wb") as f:
+            f.write(res.content)
+
+        # Extract the alt text, if any.
+        alt_match = re.search(r'alt="([^"]*)"', match.group(0))
```

Comment:
Good use of the match group!

Reply:
Thanks! I updated this in 9516f8e to use a more precise group (just the `<img ... >`).